pkg/gameserverallocations/controller_test.go

```diff
@@ -103,6 +103,7 @@ func TestControllerAllocator(t *testing.T) {
 			rec := httptest.NewRecorder()
 			err = c.processAllocationRequest(ctx, rec, r, "default")
 			assert.NoError(t, err)
+			assert.Equal(t, http.StatusCreated, rec.Code)
```

Comment:
We shouldn't block this bug fix merging, but it seems like we have a test that covers the other two case statements (using the metav1 status and returning a `http.StatusOK`).